src/pages/SignUpPage/index.tsx

```diff
@@ -0,0 +1,193 @@
+import React, { useState } from 'react';
+import { useAuth } from '../../AuthContextProvider';
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faGoogle } from "@fortawesome/free-brands-svg-icons";
+
+interface SignUpPageProps {
+  onLoginClick: () => void;
+  onSignUpSuccess: () => void;
+}
+
+export const SignUpPage: React.FC<SignUpPageProps> = ({ onLoginClick, onSignUpSuccess }) => {
```

Comment:
This component is very similar to the LoginPage. Could you combine them or at least re-use some parts?